src/classes/BGE_ConfigurationWizard_CTRL.cls

```diff
@@ -164,6 +178,117 @@ public with sharing class BGE_ConfigurationWizard_CTRL {
+
+    /*******************************************************************************************************
+    * @description Retrieves the default active fields for a new Batch or Template in a JSON String
+    */
+    public static String getDefaultActiveFields() {
+        return '[{"label":"Donation Amount",' +
+                    '"name":"Donation_Amount__c",' +
+                    '"sObjectName":"Opportunity",' +
+                    '"defaultValue":null,' +
+                    '"required":true,' +
+                    '"hide":false,' +
+                    '"sortOrder":0},' +
+                '{"label":"Donation Date",' +
+                    '"name":"Donation_Date__c",' +
+                    '"sObjectName":"Opportunity",' +
+                    '"defaultValue":null,' +
+                    '"required":false,' +
+                    '"hide":false,' +
+                    '"sortOrder":1},' +
+                '{"label":"Method",' +
+                    '"name":"Payment_Method__c",' +
+                    '"sObjectName":"Payment",' +
+                    '"defaultValue":null,' +
+                    '"required":false,' +
+                    '"hide":false,' +
+                    '"sortOrder":2},' +
+                '{"label":"Check/Reference Number",' +
+                    '"name":"Payment_Check_Reference_Number__c",' +
+                    '"sObjectName":"Payment",' +
+                    '"defaultValue":null,' +
+                    '"required":false,' +
+                    '"hide":false,' +
+                    '"sortOrder":3' +
+                '}]';
+    }
+
+    /*******************************************************************************************************
+    * @description Gets the BGEDetailsModel's default values for Batch Gift Entry
+    */
+    public static BGEDetailsModel getDefaultBGEDetailsModel() {
+        BGEDetailsModel model = new BGEDetailsModel();
+
+        model.expectedCount = 0;
+        model.expectedTotal = 0;
+        model.requireTotalMatch = false;
+        model.templateFields = new List<AvailableField>();
+        model.batchProcessSize = 50;
+        model.runOpportunityRollupsWhileProcessing = false;
+        model.contactMatchingRule = 'Firstname,Lastname,Email';
+        model.donationMatchingBehavior = BDI_DataImport_API.ExactMatchOrCreate;
+        model.donationMatchingRule = UTIL_Namespace.StrAllNSPrefix('Donation_Amount__c') + ';' + UTIL_Namespace.StrAllNSPrefix('Donation_Date__c');
+        model.donationDateRange = 0;
+        model.processUsingScheduledJob = false;
+
+        return model;
+    }
+
+    /*******************************************************************************************************
+    * @description Sets default values on BGEDetails Model to avoid returning any null values
+    */
+    private static void setDefaultValuesOnBGEDetailsModel(BGEDetailsModel model) {
+
+        if (model.expectedCount == null) {
+            model.expectedCount = 0;
```

Comment:
Should these values come from the default model? Something like:

```
BGEDetailsModel defaultModel = getDefaultBGEDetailsModel();
if (model.expectedCount == null) {
    model.expectedCount = defaultModel.expectedCount;
}
```

You could probably even iterate through fields on either model instead of referencing each one individually.

Reply:
that is....a much more elegant implementation. on it.